setup.py

```diff
@@ -18,19 +21,37 @@
     description="Annotated Data.",
     long_description=Path("README.rst").read_text("utf-8"),
     url="http://github.com/theislab/anndata",
-    author=__author__,
-    author_email=__email__,
+    author=metadata["author"],
+    author_email=metadata["author-email"],
     license="BSD-3-Clause",
     install_requires=[
         l.strip() for l in Path("requirements.txt").read_text("utf-8").splitlines()
     ],
     extras_require=dict(
-        docs=[
-            "sphinx",
-            "sphinx_rtd_theme",
-            "sphinx_autodoc_typehints",
+        dev=["setuptools_scm", "pytoml", "black"],
+        doc=[
+            # Sphinx 2 has nicer looking sections
+            "sphinx>=2.0.1",
+            "sphinx-rtd-theme",
+            "sphinx-autodoc-typehints>=1.11.0",
+            "sphinx_issues",
             "scanpydoc>=0.5",
-        ]
+            'typing_extensions; python_version < "3.8"',
+        ],
+        test=[
+            "loompy>=3.0.5",
+            "pytest>=4.6",
+            "pytest-cov>=2.10",
+            "codacy-coverage",
+            "docutils",  # for rst2html.py
+            "zarr",
+            "matplotlib",
+            "sklearn",
+            "xlrd",
+            "joblib",
+            "boltons",
+            "scanpy",
+        ],
```

Comment:
Could these all be in a config file instead of `setup.py`? I'm thinking something like [this](https://github.com/jaraco/skeleton/blob/master/setup.cfg), but as entries in `pyproject.toml` and not tool specific.